File: angular-Cinema-frontend/src/app/models/projection.ts
import { Salle } from "./Salle";
import { Time } from "./Time";
import { Film } from "./film";

export class Projection {
        map(arg0: (i: any) => void) {
          throw new Error('Method not implemented.');
        }
        public id!: number;
        public dateProjection! : Date;
        public salle! : Salle;
        public time! : Time;
        public film! : Film
}

File: spring-boot-cinema/src/main/java/com/luv2code/cinema/entity/Projection.java
package com.luv2code.cinema.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Table(name="projection")
@Data
public class Projection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "date_projection")
    private Date dateProjection;

    @Column(name = "nbPlaceDispo")
    private int nbPlaceDispo;
    @ManyToOne
    @JoinColumn(name = "film_id", nullable = false)
    private Film film;

    @ManyToOne
    @JoinColumn(name = "salle_id", nullable = false)
    private Salle salle;

    @ManyToOne
    @JoinColumn(name = "time_id", nullable = false)
    private Time time;

    public Projection() {
    }

    public Projection(Long id, Date dateProjection, Film film, Salle salle, Time time) {
        this.id = id;
        this.dateProjection = dateProjection;
        this.film = film;
        this.salle = salle;
        this.time = time;
    }
}
